src/ngMock/angular-mocks.js

```diff
@@ -2006,6 +2009,17 @@ function MockXhr() {
   };
 
   this.abort = angular.noop;
+
+  this.$$events = {};
+  this.addEventListener = function(name, listener) {
+    if (angular.isUndefined(this.$$events[name])) this.$$events[name] = [];
+    this.$$events[name].push(listener);
+  };
```

Comment:
I agree this is not optimal.
The `MockXhr` is used in two different scenarios:
* In the `$httpBackend` tests we instantiate `MockXhr` objects and pass them through to the `$httpBackend` service, which does call `addEventListener`. This allows us to test the "real" use of the `$httpBackend` service.
* In the `$http` tests, we have a mock `$httpBackend` service (from `ngMock`), which happens to hold its own instance of `MockXhr`. In these tests we never call `addEventListener` and I just chose to use the same `$$events` object for holding our mock event handlers for the test